src/pfe/portal/routes/projects/remoteBind.route.js

```diff
@@ -270,8 +270,13 @@ async function uploadEnd(req, res) {
       await cwUtils.copyProject(pathToTempProj, path.join(project.workspace, project.directory), getMode(project));
 
       if (project.injectMetrics) {
+        const projectDir = path.join(project.workspace, project.directory);
+        if (modifiedList.includes("pom.xml")) {
```

Comment:
This checks for a pom.xml but most of the other functions are project type neutral. This issue probably affects NodeJS as well.
Either we need a function on metricsService that can return the name (or names) of the files that trigger reinjection so we can decide correctly here or we just always re-inject. We would only need to restore from the backup when injection was disabled.